contentcuration/contentcuration/frontend/channelEdit/components/AddRelatedResourcesModal.vue

```diff
@@ -0,0 +1,231 @@
+        </VToolbarItems>
+      </VToolbar>
+
+      <p class="mt-4 ml-2">
+        {{ $tr('resourcesDisplayedText') }}
+        <span class="font-weight-bold">&apos;{{ targetNodeTitle }}&apos;</span>
+      </p>
+
+      <NodeTreeNavigation
+        v-if="selectedNodeId"
+        v-model="selectedNodeId"
+        :channelId="currentChannelId"
+      >
+        <VListTile
+          slot="child"
+          :key="childNode.id"
+          slot-scope="{ childNode }"
+          :class="listItemClasses(childNode)"
+          @click="onListItemClick(childNode)"
+        >
+          <VListTileAction>
+            <ContentNodeIcon
+              v-if="childNode.kind"
+              :kind="childNode.kind"
+              :size="20"
+            />
+          </VListTileAction>
+
+          <VListTileContent>
+            <VListTileTitle>
+              <VTooltip
+                right
+                :disabled="!isListItemDisabled(childNode)"
+              >
+                <template v-slot:activator="{ on }">
+                  <span v-on="on">{{ childNode.title }}</span>
+                </template>
+                <span>{{ listItemTooltip(childNode) }}</span>
+              </VTooltip>
+            </VListTileTitle>
+          </VListTileContent>
+
+          <VListTileAction v-if="displayAddButton(childNode)">
+            <VBtn
+              flat
+              color="primary"
+              class="font-weight-bold"
+              @click.stop.prevent="onAddStepClick(childNode)"
+            >
+              {{ $tr('addStepBtnLabel') }}
+            </VBtn>
+          </VListTileAction>
+        </VListTile>
+      </NodeTreeNavigation>
+    </VCard>
+  </VDialog>
+
+</template>
+
+<script>
+
+  import { mapGetters, mapActions } from 'vuex';
+
+  import { RouterNames } from '../constants';
+  import NodeTreeNavigation from './NodeTreeNavigation';
+  import { ContentNodeKind } from 'shared/constants';
+  import { TabNames } from 'edit_channel/uploader/constants';
+
+  import ContentNodeIcon from 'shared/views/ContentNodeIcon';
+
+  export default {
+    name: 'AddRelatedResourcesModal',
+    components: {
+      ContentNodeIcon,
+      NodeTreeNavigation,
+    },
+    props: {
+      targetNodeId: {
+        type: String,
+        required: true,
+      },
+      toolbarTitle: {
+        type: String,
+        required: true,
+      },
+      selectedAsPreviousStepTooltip: {
+        type: String,
+        required: true,
+      },
+      selectedAsNextStepTooltip: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        selectedNodeId: null,
+      };
+    },
+    computed: {
+      ...mapGetters('currentChannel', ['currentChannel']),
+      ...mapGetters('contentNode', [
+        'getContentNode',
+        'getContentNodeParents',
+        'isPreviousStep',
+        'isNextStep',
+      ]),
+      currentChannelId() {
+        return this.currentChannel ? this.currentChannel.id : null;
```

Comment:
I checked it in more detail and current channel ID is available on channel edit SPA load. https://github.com/learningequality/studio/blob/vue-refactor/contentcuration/contentcuration/templates/channel_edit.html

It's rather a general precaution of handling asynchronous data - especially when it comes to getters, it can happen that a getter returns `undefined`/`null` only to return data a few milliseconds later and that generates "false" runtime errors. In this way, I also try to keep logic of views decoupled and prepared for any updates of logic related to fetching data (for example right now we can rely on having current channel ID available because of our multiple SPA architecture though that might not be the case in the future). Anyway, I've noticed that although I added this check, I don't treat missing current channel ID before using it in other places so it's good for nothing in this case :))

When checking this, I found that we have current channel ID available in state https://github.com/learningequality/studio/blob/vue-refactor/contentcuration/contentcuration/frontend/channelEdit/vuex/currentChannel/index.js so I think it will be best to simply reach directly there.